skorch/net.py

```diff
@@ -1285,6 +1272,55 @@ def get_params(self, deep=True, **kwargs):
         params.update(params_cb)
         return params
 
+    def check_kwargs(self, kwargs):
+        """Check argument names passed at initialization.
```

Comment:
For our future selves, maybe add a small example on how `iterator_train_shuffle` will raise a `TypeError` that will suggest `iterator_train__shuffle`.